### PR TITLE
feat: [IOPID-1506,IOPID-1747,IOPID-1788,IOPID-1536] - Share data screen MP, new DS integration

### PR DESCRIPTION
**Pull request by ChrisMattew**
## Short description
Integration of new DS in Share data screen component, addition of new MP event

## List of changes proposed in this pull request
- Integrate new DS
- Adjust link to security policy
- Tracked new `TRACKING_INFO` event on MixPanel

## Screens

<details><summary>iOS</summary>

|Onboarding|Profile|
|-|-|
|<video  src="https://github.com/pagopa/io-app/assets/94987118/8cce6870-0219-4977-9549-15b97f28ee8b" />|<video src="https://github.com/pagopa/io-app/assets/94987118/f59d5e7e-0dbc-4db1-8f5d-084a43974dac" />|

|A11Y Onboarding|A11Y Profile|
|-|-|
|<video  src="https://github.com/pagopa/io-app/assets/94987118/d4576218-3b74-40be-a6d0-c2a4c2950d81" />|<video src="https://github.com/pagopa/io-app/assets/94987118/38ac47dd-8211-401c-8ec4-bea8e5c01710" />|

</details>

<details><summary>Android</summary>

|Onboarding|Profile|
|-|-|
|<video  src="https://github.com/pagopa/io-app/assets/94987118/4551d582-12cb-41b8-82dd-7bb88280b48b" />|<video src="https://github.com/pagopa/io-app/assets/94987118/b0d770c2-b520-4155-821a-22574c227aa8" />|

|A11Y Onboarding|A11Y Profile|
|-|-|
|<video  src="https://github.com/pagopa/io-app/assets/94987118/e002008d-8bf2-4e65-a7a7-96b88cc4254f" />|<video src="https://github.com/pagopa/io-app/assets/94987118/9141dc08-4a81-4b13-a5bd-26a664b72968" />|

</details>

## How to test
There are two scenarios in which test the new `ShareDataScreen`: from `Onboarding` flow and from `Profile > Privacy Policy and Terms of Service > Share usage data`.

### CODE REVIEW

**Comment by ChrisMattew** [comment]
@mariateresaventura copy review needed 🙏🏼

File: ts/screens/profile/ShareDataScreen.tsx [hunk 1]
```diff
@@ -1,121 +1,138 @@
...
+    showUpdatedPreferencesToastMessage();
   });
-  const isMixpanelEnabled = props.isMixpanelEnabled ?? true;
+
+  const dontShareDataLabel = I18n.t(
+    "profile.main.privacy.shareData.screen.cta.dontShareData"
+  );
+  const shareDataLabel = I18n.t(
+    "profile.main.privacy.shareData.screen.cta.shareData"
+  );
```
Comment (shadowsheep1) [inline ts/screens/profile/ShareDataScreen.tsx:59]: In the [JIRA](https://pagopa.atlassian.net/browse/IOPID-1536) task the screenshot show an **outline** button, while in [FIGMA](https://www.figma.com/design/zaqsSaM3iGl7iSaGoywYNK/Sezione-Profilo?node-id=1831-29141&t=BDQqp7lMhcRuxrUO-4) we have a **solid** button for the deactivation too. Is FIGMA the right one, am I right?

Comment (ChrisMattew) [reply ts/screens/profile/ShareDataScreen.tsx:59]: You're right, the Figma is the right one!

**Comment by mariateresaventura** [comment]
> @mariateresaventura copy review needed 🙏🏼

Done!

File: ts/screens/profile/components/ShareDataComponent/ShareDataFeatureInfos.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      <LegacyMarkdown>
+        {I18n.t("profile.main.privacy.shareData.whyBottomSheet.body")}
+      </LegacyMarkdown>
+    </View>
+  </View>
+);
+
+const AnalyticsFeatureInfo = ({ trackAction }: FeatureProps) => {
+  const bodyRef = useRef<View>(null);
+  const { present, bottomSheet } = useLegacyIOBottomSheetModal(
```
[the remark this answers is not in the document]
Comment (ChrisMattew) [reply ts/screens/profile/components/ShareDataComponent/ShareDataFeatureInfos.tsx:51]: Good catch, I will replace it!

File: ts/screens/profile/components/ShareDataComponent/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      <VSpacer size={16} />
+      <AnalyticsFeatureInfo trackAction={trackAction} />
+      <VSpacer size={24} />
+      <SecurityFeatureInfo trackAction={trackAction} />
+      <VSpacer size={24} />
+      <GDPRFeatureInfo trackAction={trackAction} />
+      <VSpacer size={32} />
+      <Body
+        accessibilityRole="link"
+        onPress={handleOnPress}
```
[the remark this answers is not in the document]
Comment (ChrisMattew) [reply ts/screens/profile/components/ShareDataComponent/index.tsx:33]: It's a remnant of the old implementation. I missed it, well done!

File: ts/screens/profile/components/ShareDataComponent/ShareDataFeatureInfos.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChrisMattew) [reply ts/screens/profile/components/ShareDataComponent/ShareDataFeatureInfos.tsx]: Done!

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**